### ISSUE TITLE
[Bug] Conjugate gradients algorithm Solve with JAX uses a deprecated function

### ISSUE THREAD

**Issue by adam-hartshorne** [wilson-labs/cola#111]
# 🐛 Bug

If one attempts to use the CG algorithm to solve for an inverse, it will call  host_callback.id_tap(construct_info, None) inside of jax_tqdm.py.  However, jax.experimental.host_callback has been deprecated since March 2024 and is now no longer supported. See https://github.com/jax-ml/jax/issues/20385

**Comment by AndPotap** [comment]
@adam-hartshorne, I'm looking into this.
